src/Symfony/Component/Workflow/Workflow.php

```diff
@@ -22,249 +20,31 @@
  * @author Fabien Potencier <fabien@symfony.com>
  * @author Grégoire Pineau <lyrixx@lyrixx.info>
  */
-class Workflow
+class Workflow extends PetriNet
 {
-    private $definition;
-    private $markingStore;
-    private $dispatcher;
-    private $name;
-
-    public function __construct(Definition $definition, MarkingStoreInterface $markingStore, EventDispatcherInterface $dispatcher = null, $name = 'unnamed')
-    {
-        $this->definition = $definition;
-        $this->markingStore = $markingStore;
-        $this->dispatcher = $dispatcher;
-        $this->name = $name;
+    public function __construct(
+        Definition $definition,
+        MarkingStoreInterface $markingStore,
+        EventDispatcherInterface $dispatcher = null,
+        $name = 'unnamed'
+    ) {
+        parent::__construct($definition, $markingStore, $dispatcher, $name);
 
         // If the marking can contain only one place, we should control the definition
         if ($markingStore instanceof UniqueTransitionOutputInterface) {
+            /** @var Transition $transition */
```

Comment:
this is useless if we properly document the return type of ``getTransitions`` as ``Transition[]``, so please remove this inline phpdoc